### DISCUSSION TITLE
Input Shape for RGB

### DISCUSSION THREAD

**Discussion by rt-a** [kymatio/kymatio#755] [Q&A]
Hi,

for an university project I'm trying to implement a Scattering Network as feature extractor in a faster-rcnn object detection network. 
I used [this implementation ](https://github.com/you359/Keras-FasterRCNN/) and used one of kymatio's examples as feature extractor.

`inputs = Input(shape=(width, height))
`
`x = Scattering2D(J=3, L=8)(inputs)
`

They trained and tested it on VOC Dataset which uses RGB data. How do I use kymatio for RGB data? I couldn't find anything in the documentation. There seems to be a way because they  compare computation time for 128 256x256 RGB images for different gpu's. Thanks for your help.

**Comment by lostanlen** [comment]
Hello @rt-a , you can pass an array of shape `(3, width, height)` to `Scattering2D`.
See: https://github.com/kymatio/kymatio/blob/master/kymatio/scattering2d/frontend/base_frontend.py#L162

**Reply by OverLordGoldDragon** [reply]
And more generally as `(3*samples, width, height)` then `.reshape(samples, 3, width, height)` as input to an NN. If `channels_last` then a bit trickier: `.reshape(samples, 3, width, height).transpose(0, 2, 3, 1)`. These assume you fed to scattering as `[R0, G0, B0, R1, G1, B1, ...]`.

**Reply by rt-a** [reply]
Hi, thanks for the quick help.
The faster-rcnn is implemented in Keras and with `from kymatio.keras import Scattering2D` I can't pass `shape` because it doesn't know this keyword argument.
If I use `kymatio.tensorflow` to pass the input_shape to `Scattering2D` like this: 
`x = Scattering2D(J=3, L=12, shape=input_shape)(img_input)`
I get an error `too many values to unpack (expected 2)` 
Now in the faster-rcnn implementation they use `input_shape = (None, None, 3)` as Shape and `img_input`  is a `<KerasTensor: shape=(None, None, None, 3)`

I reduced `input_shape` to 2 values and I don't get the error.
Am I missing something?

I use TF 2.5.0 and Kymatio 0.2.1
